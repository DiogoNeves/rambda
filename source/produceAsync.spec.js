import { delay } from './delay'
import { produceAsync } from './produceAsync'

test('happy', async () => {
  const fn = produceAsync({
    foo : async x => {
      await delay(100)

      return `${ x }_ZEPPELIN`
    },
    bar : inputArgument => inputArgument === 5,
  })
  const expected = {
    foo : 'LED_ZEPPELIN',
    bar : false,
  }

  const result = await fn('LED')
  expect(result).toEqual(expected)
})

test('with error', async () => {
  const fn = produceAsync({
    foo : async x => {
      await delay(100)
      throw new Error(`${ x }_ZEPPELIN`)
    },
    bar : inputArgument => inputArgument === 5,
  })

  try {
    await fn('LED')
    expect(1).toBe(2)
  } catch (e){
    expect(e.message).toBe('LED_ZEPPELIN')
  }
})
