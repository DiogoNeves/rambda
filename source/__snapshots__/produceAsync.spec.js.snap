// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile Number 1 | Number 1 1`] = `
Object {
  "error": Object {
    "message": "produceFn(...).then is not a function",
    "type": "TypeError",
  },
  "inputs": Array [
    1,
    1,
  ],
  "result": "PENDING",
}
`;

exports[`profile Number 1 | Object {} 1`] = `
Object {
  "error": Object {
    "message": "produceFn(...).then is not a function",
    "type": "TypeError",
  },
  "inputs": Array [
    1,
    Object {},
  ],
  "result": "PENDING",
}
`;

exports[`profile Number 1 | Promise  1`] = `
Object {
  "error": Object {
    "message": "produceFn(...).then is not a function",
    "type": "TypeError",
  },
  "inputs": Array [
    1,
    Promise {},
  ],
  "result": "PENDING",
}
`;

exports[`profile Number 1 | RegExp /foo/g 1`] = `
Object {
  "error": Object {
    "message": "produceFn(...).then is not a function",
    "type": "TypeError",
  },
  "inputs": Array [
    1,
    /foo/g,
  ],
  "result": "PENDING",
}
`;

exports[`profile Object {"a":""} | Number 1 1`] = `
Object {
  "error": "PENDING",
  "inputs": Array [
    Object {
      "a": [Function],
    },
    1,
  ],
  "result": Object {
    "a": 2,
  },
}
`;

exports[`profile Object {"a":""} | Object {} 1`] = `
Object {
  "error": "PENDING",
  "inputs": Array [
    Object {
      "a": [Function],
    },
    Object {},
  ],
  "result": Object {
    "a": "[object Object]1",
  },
}
`;

exports[`profile Object {"a":""} | Promise  1`] = `
Object {
  "error": "PENDING",
  "inputs": Array [
    Object {
      "a": [Function],
    },
    Promise {},
  ],
  "result": Object {
    "a": "[object Promise]1",
  },
}
`;

exports[`profile Object {"a":""} | RegExp /foo/g 1`] = `
Object {
  "error": "PENDING",
  "inputs": Array [
    Object {
      "a": [Function],
    },
    /foo/g,
  ],
  "result": Object {
    "a": "/foo/g1",
  },
}
`;

exports[`profile Object {"a":"[1]"} | Number 1 1`] = `
Object {
  "error": Object {
    "message": "conditions[prop] is not a function",
    "type": "TypeError",
  },
  "inputs": Array [
    Object {
      "a": Array [
        1,
      ],
    },
    1,
  ],
  "result": "PENDING",
}
`;

exports[`profile Object {"a":"[1]"} | Object {} 1`] = `
Object {
  "error": Object {
    "message": "conditions[prop] is not a function",
    "type": "TypeError",
  },
  "inputs": Array [
    Object {
      "a": Array [
        1,
      ],
    },
    Object {},
  ],
  "result": "PENDING",
}
`;

exports[`profile Object {"a":"[1]"} | Promise  1`] = `
Object {
  "error": Object {
    "message": "conditions[prop] is not a function",
    "type": "TypeError",
  },
  "inputs": Array [
    Object {
      "a": Array [
        1,
      ],
    },
    Promise {},
  ],
  "result": "PENDING",
}
`;

exports[`profile Object {"a":"[1]"} | RegExp /foo/g 1`] = `
Object {
  "error": Object {
    "message": "conditions[prop] is not a function",
    "type": "TypeError",
  },
  "inputs": Array [
    Object {
      "a": Array [
        1,
      ],
    },
    /foo/g,
  ],
  "result": "PENDING",
}
`;

exports[`profile Object {"a":1} | Number 1 1`] = `
Object {
  "error": Object {
    "message": "conditions[prop] is not a function",
    "type": "TypeError",
  },
  "inputs": Array [
    Object {
      "a": 1,
    },
    1,
  ],
  "result": "PENDING",
}
`;

exports[`profile Object {"a":1} | Object {} 1`] = `
Object {
  "error": Object {
    "message": "conditions[prop] is not a function",
    "type": "TypeError",
  },
  "inputs": Array [
    Object {
      "a": 1,
    },
    Object {},
  ],
  "result": "PENDING",
}
`;

exports[`profile Object {"a":1} | Promise  1`] = `
Object {
  "error": Object {
    "message": "conditions[prop] is not a function",
    "type": "TypeError",
  },
  "inputs": Array [
    Object {
      "a": 1,
    },
    Promise {},
  ],
  "result": "PENDING",
}
`;

exports[`profile Object {"a":1} | RegExp /foo/g 1`] = `
Object {
  "error": Object {
    "message": "conditions[prop] is not a function",
    "type": "TypeError",
  },
  "inputs": Array [
    Object {
      "a": 1,
    },
    /foo/g,
  ],
  "result": "PENDING",
}
`;

exports[`profile String  | Number 1 1`] = `
Object {
  "error": Object {
    "message": "produceFn(...).then is not a function",
    "type": "TypeError",
  },
  "inputs": Array [
    "",
    1,
  ],
  "result": "PENDING",
}
`;

exports[`profile String  | Object {} 1`] = `
Object {
  "error": Object {
    "message": "produceFn(...).then is not a function",
    "type": "TypeError",
  },
  "inputs": Array [
    "",
    Object {},
  ],
  "result": "PENDING",
}
`;

exports[`profile String  | Promise  1`] = `
Object {
  "error": Object {
    "message": "produceFn(...).then is not a function",
    "type": "TypeError",
  },
  "inputs": Array [
    "",
    Promise {},
  ],
  "result": "PENDING",
}
`;

exports[`profile String  | RegExp /foo/g 1`] = `
Object {
  "error": Object {
    "message": "produceFn(...).then is not a function",
    "type": "TypeError",
  },
  "inputs": Array [
    "",
    /foo/g,
  ],
  "result": "PENDING",
}
`;
